tools/src/main/scala/com/nvidia/spark/rapids/tool/ThreadFactoryBuilder.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tool
+
+import java.util.concurrent.{Executors, ThreadFactory}
+import java.util.concurrent.atomic.AtomicLong
+
+// This is similar to Guava ThreadFactoryBuilder
+// Avoid to use Guava as it is a messy dependency in practice.
+// This is copied from the common module
```

Comment:
We are already using the shade plugin to include the common module.  So I see this more as a problem of not using a dependency-reduced pom rather than the need to copy code.  For example, the pom probably also says it needs the scallop jar as a dependency despite the fact that we are using the shade plugin to include it.  This PR removed the need for the common module but the config for the shade plugin still requests its inclusion.

Personally I'd rather see us fix the pom, using a dependency-reduced pom to publish rather than shade _and_ still list shaded dependencies as it is doing today.